tss-esapi/src/interface_types/algorithm.rs

```diff
@@ -335,6 +337,32 @@ impl TryFrom<TPMI_ALG_ASYM> for AsymmetricAlgorithm {
     }
 }
 
+/// Enum representing the asymmetric algorithm interface type with specific properties.
+///
+/// # Details
+/// Use this instead of [AsymmetricAlgorithm].
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum AsymmetricAlgorithmSelection {
+    Rsa(RsaKeyBits),
+    Ecc(EccCurve),
+}
+
+/// The conversion assumes for RSA 2048 bit size and for ECC the Nist P256 curve,
+/// which matches the defaults in tpm2-tools.
+impl TryFrom<AsymmetricAlgorithm> for AsymmetricAlgorithmSelection {
+    type Error = Error;
+
+    fn try_from(value: AsymmetricAlgorithm) -> std::result::Result<Self, Self::Error> {
+        match value {
+            AsymmetricAlgorithm::Rsa => Ok(AsymmetricAlgorithmSelection::Rsa(RsaKeyBits::Rsa2048)),
+            AsymmetricAlgorithm::Ecc => Ok(AsymmetricAlgorithmSelection::Ecc(EccCurve::NistP256)),
+            AsymmetricAlgorithm::Null => {
+                Err(Error::local_error(WrapperErrorKind::UnsupportedParam))
+            }
+        }
+    }
+}
+
```

Comment:
I think it would be nice with a conversion from ```AsymmetricAlgorithmSelection``` to a ```NvIndexTpmHandle``` so that one can avoid those constants in ek.rs.

Reply:
`AsymmetricAlgorithmSelection` is also used for AKs and in that context the conversion to an EK certificate index does not really make sense, but indeed it would make the conversion nicer.

Reply:
hrrm no that is true. In order to make that work you have to create an extra type (something like ```EkCertificateNvIndex```) and then implement the desired conversions for it. But I am not sure if it is worth the trouble.